runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java

```diff
@@ -209,10 +206,10 @@ private void updatePresenceWithLatestEvent()
 				break;
 			case TOTAL:
 				// We are tracking total time spent instead of per activity time so try to find
-				// IN_GAME or IN_MENU event as this indicates start of tracking and find last updated one
-				// to determine if we are in game or menu
+				// root event as this indicates start of tracking and find last updated one
+				// to determine correct state we are in
 				startTime = events.stream()
-					.filter(e -> e.type == DiscordGameEventType.IN_GAME || e.type == DiscordGameEventType.IN_MENU)
+					.filter(e -> e.getType().isRoot())
 					.sorted((a, b) -> b.getUpdated().compareTo(a.getUpdated()))
 					.map(EventWithTime::getStart)
 					.findFirst()
```

Comment:
Assuming there's only 1 root event in events list at any given time, this could be simplified.

Reply:
Well in-menu theoretically can be active at same time as in-game timer

Reply:
Yeah, this is just to clarify which section of code I was referring to in the above post. Although both can be in the events list, is it possible for the player to actually be both in-menu and in-game at the same time?